File: src/main/java/com/darrenfinch/feastyweb/food/Food.java
package com.darrenfinch.feastyweb.food;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.Hibernate;

import java.util.Objects;

@Getter
@Setter
@ToString
@RequiredArgsConstructor
@Entity
@Table(name = "food")
public class Food {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name="userId")
    private Long userId;

    @Column(name = "title")
    private String title;

    @Column(name = "quantity")
    private double quantity;

    @Column(name = "unit")
    private String unit;

    @Column(name = "calories")
    private int calories;

    @Column(name = "fats")
    private int fats;

    @Column(name = "carbs")
    private int carbs;

    @Column(name = "proteins")
    private int proteins;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Food food = (Food) o;
        return id != null && Objects.equals(id, food.id);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: new-backend-app/Models/Food.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace new_backend.Models
{
    public class Food
    {
        [Required]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public virtual long Id { get; set; }

        public virtual long UserId { get; set; }

        [Required]
        public virtual string Title { get; set; }

        [Required]
        public virtual double Quantity { get; set; }

        [Required]
        public virtual string Unit { get; set; }

        [Required]
        public virtual int Calories { get; set; }

        [Required]
        public virtual int Fats { get; set; }

        [Required]
        public virtual int Carbs { get; set; }

        [Required]
        public virtual int Proteins { get; set; }

        public Food()
        {
        }
    }
}